crates/jrsonnet-macros/src/lib.rs

```diff
@@ -538,40 +545,79 @@ impl TypedField {
 		})
 	}
 	fn expand_parse(&self) -> TokenStream {
+		if self.is_option {
+			self.expand_parse_optional()
+		} else {
+			self.expand_parse_mandatory()
+		}
+	}
+
+	fn expand_parse_optional(&self) -> TokenStream {
 		let ident = &self.ident;
 		let ty = &self.ty;
+
+		// optional flatten is handled in same way as serde
 		if self.attr.flatten {
-			// optional flatten is handled in same way as serde
-			return if self.is_option {
-				quote! {
-					#ident: <#ty as TypedObj>::parse(&obj).ok(),
-				}
-			} else {
-				quote! {
-					#ident: <#ty as TypedObj>::parse(&obj)?,
-				}
+			return quote! {
+				#ident: <#ty as TypedObj>::parse(&obj).ok(),
 			};
-		};
+		}
 
 		let name = self.name().unwrap();
-		let value = if self.is_option {
-			quote! {
-				if let Some(value) = obj.get(#name.into())? {
-					Some(<#ty as Typed>::from_untyped(value)?)
-				} else {
-					None
-				}
+		let aliases = &self.attr.aliases;
+
+		let value = quote! {
+			if let Some(__value) = obj.get(#name.into())? {
+				Some(<#ty as Typed>::from_untyped(__value)?)
+			} #(else if let Some(__value) = obj.get(#aliases) {
+				Some(<#ty as Typed>::from_untyped(__value)?)
+			})* else {
+				None
 			}
+		};
+
+		quote! {
+			#ident: #value,
+		}
+	}
+
+	fn expand_parse_mandatory(&self) -> TokenStream {
+		let ident = &self.ident;
+		let ty = &self.ty;
+
+		// optional flatten is handled in same way as serde
+		if self.attr.flatten {
+			return quote! {
+				#ident: <#ty as TypedObj>::parse(&obj)?,
+			};
+		}
+
+		let name = self.name().unwrap();
+		let aliases = &self.attr.aliases;
+
+		let error_text = if aliases.is_empty() {
+			// clippy does not understand name variable usage in quote! macro
+			#[allow(clippy::redundant_clone)]
+			name.clone()
 		} else {
-			quote! {
-				<#ty as Typed>::from_untyped(obj.get(#name.into())?.ok_or_else(|| ErrorKind::NoSuchField(#name.into(), vec![]))?)?
+			format!("{name} (alias {})", aliases.join(", "))
+		};
+
+		let value = quote! {
+			if let Some(__value) = obj.get(#name.into())? {
```

Comment:
<#ty as Typed>::from_untyped call might be moved to outside, 
```rust
<#ty as Typed>::from_untyped(if let Some(...) = ...)
```

Reply:
I know. It was done intentionally. I think that moving `from_untyped` makes the code less readable.

Reply:
Yet greately inflates compilation time, as compiler won't be able to merge those calls

Reply:
It might be split for better readability

let value = if let ...;
<#ty as Typed>::from_untyped(value)